docs/tutorials/dev_add_test_group.md

```diff
@@ -0,0 +1,2218 @@
+        test_group : polaris.ocean.tests.baroclinic_channel.BaroclinicChannel
+            The test group that this test case belongs to
+
+        resolution : float
+            The resolution of the test case in km
+
+        name : str
+            The name of the test case
+        """
+        self.resolution = resolution
+        if resolution >= 1.:
+            res_str = f'{resolution:g}km'
+        else:
+            res_str = f'{resolution * 1000.:g}m'
+        subdir = os.path.join(res_str, name)
+        super().__init__(test_group=test_group, name=name,
+                         subdir=subdir)
+
+        self.add_step(
+            InitialState(test_case=self, resolution=resolution))
+
+    def validate(self):
+        """
+        Compare ``temperature``, ``salinity`` and ``layerThickness`` from the
+        initial condition with a baseline if one was provided
+        """
+        super().validate()
+        variables = ['temperature', 'salinity', 'layerThickness']
+        compare_variables(test_case=self, variables=variables,
+                          filename1='initial_state/initial_state.nc')
+
+```
+
+Now, we'll make `Default` descend from `BaroclinicChannelTestCase` and remove
+the redundant pieces.  Here's what's left:
+
+```python
+from polaris.ocean.tests.baroclinic_channel import BaroclinicChannelTestCase
+from polaris.ocean.tests.baroclinic_channel.forward import Forward
+from polaris.ocean.tests.baroclinic_channel.viz import Viz
+from polaris.validate import compare_variables
+
+
+class Default(BaroclinicChannelTestCase):
+    """
+    The default test case for the baroclinic channel test group simply creates
+    the mesh and initial condition, then performs a short forward run on 4
+    cores.
+    """
+
+    def __init__(self, test_group, resolution):
+        """
+        Create the test case
+
+        Parameters
+        ----------
+        test_group : polaris.ocean.tests.baroclinic_channel.BaroclinicChannel
+            The test group that this test case belongs to
+
+        resolution : float
+            The resolution of the test case in km
+        """
+        super().__init__(test_group=test_group, resolution=resolution,
+                         name='default')
+
+        self.add_step(
+            Forward(test_case=self, ntasks=4, min_tasks=4, openmp_threads=1,
+                    resolution=resolution, run_time_steps=3))
+
+        self.add_step(
+            Viz(test_case=self))
+
+    def validate(self):
+        """
+        Compare ``temperature``, ``salinity``, ``layerThickness`` and
+        ``normalVelocity`` in the ``forward`` step with a baseline if one was
+        provided.
+        """
+        super().validate()
+        variables = ['temperature', 'salinity', 'layerThickness',
+                     'normalVelocity']
+        compare_variables(test_case=self, variables=variables,
+                          filename1='forward/output.nc')
+
+```
+
+## Adding another test case
```

Comment:
```suggestion
## Adding a parameter study
```

My inclination would be to keep this section (and down to L2126) short to just the bits of code that are relevant to setting up a parameter study, but since you've already written it, you could point out the relevant bits if we are going through it together or leave it as future reading.

Reply:
I don't feel too strongly about keeping the longer version of this, so feel free to shorten it (or I can).